ballerina/modules/parser/tests/parser_tests.bal

```diff
@@ -509,6 +509,94 @@ isolated function testVariables() returns error? {
+    test:assertEquals(variableDefinition.name, "age");
+    test:assertEquals(variableDefinition.kind, "Int!");
+    FieldNode fieldNode = operationNode.getFields()[0];
+    test:assertEquals(fieldNode.getName(), "profile");
+    ArgumentNode argumentNode = fieldNode.getArguments()[0];
+    test:assertEquals(argumentNode.getName(), "userName");
+    test:assertEquals(argumentNode.isVariableDefinition(), true);
+    test:assertEquals(argumentNode.getVariableName(), "name");
+    argumentNode = fieldNode.getArguments()[1];
+    test:assertEquals(argumentNode.getName(), "userAge");
+    test:assertEquals(argumentNode.isVariableDefinition(), true);
+    test:assertEquals(argumentNode.getVariableName(), "age");
+}
+
+@test:Config {
+    groups: ["variables", "list", "parser"]
+}
+isolated function testListTypeVariables() returns error? {
+    string document = "query getId($name: [String!]!) { profile(userName:$name) { id } }";
+    Parser parser = new(document);
+    DocumentNode documentNode = check parser.parse();
+    test:assertEquals(documentNode.getOperations().length(), 1);
+    OperationNode operationNode = documentNode.getOperations()[0];
+    test:assertEquals(operationNode.getVaribleDefinitions().length(), 1);
+    VariableDefinition variableDefinition = <VariableDefinition> operationNode.getVaribleDefinitions()["name"];
+    test:assertEquals(variableDefinition.name, "name");
+    test:assertEquals(variableDefinition.kind, "[String!]!");
+    FieldNode fieldNode = operationNode.getFields()[0];
+    test:assertEquals(fieldNode.getName(), "profile");
+    ArgumentNode argumentNode = fieldNode.getArguments()[0];
+    test:assertEquals(argumentNode.getName(), "userName");
+    test:assertEquals(argumentNode.isVariableDefinition(), true);
+    test:assertEquals(argumentNode.getVariableName(), "name");
+}
+
+@test:Config {
+    groups: ["variables", "list", "parser"]
+}
+isolated function testInvalidListTypeVariableMissingOpenBracket() returns error? {
+    string document = "query getId($name: String!]!) { profile(userName:$name) { id } }";
+    Parser parser = new(document);
+    DocumentNode|Error result = parser.parse();
+    test:assertTrue(result is InvalidTokenError);
+    InvalidTokenError err = <InvalidTokenError>result;
+    string expectedMessage = string`Syntax Error: Expected "$", found "]".`;
+    test:assertEquals(err.message(), expectedMessage);
+}
+
+@test:Config {
+    groups: ["variables", "list", "parser"]
+}
+isolated function testInvalidListTypeVariableMissingCloseBracket() returns error? {
+    string document = "query getId($name: [String![) { profile(userName:$name) { id } }";
+    Parser parser = new(document);
+    DocumentNode|Error result = parser.parse();
+    test:assertTrue(result is InvalidTokenError);
+    InvalidTokenError err = <InvalidTokenError>result;
+    string expectedMessage = string`Syntax Error: Expected "]", found "[".`;
+    test:assertEquals(err.message(), expectedMessage);
```

Comment:
Shouldn't we assert the error location as well? As a practice, when there's an error, let's check the error location as well.